### PR TITLE
[13.0][IMP] Add selection: choose which currency should be used in risk

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Forward-port of https://github.com/OCA/credit-control/pull/83.

Please, use **merge major**!

### CODE REVIEW

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional review works as expected.

**Comment by pedrobaeza** [comment]
Trying on runbot, I see something weird:

Peek 04-09-2020 08-59

Changing the currency in the contact, I would expect the amount of opened operations to change according that currency.

**Comment by MiquelRForgeFlow** [comment]
@pedrobaeza that's not weird. You are editing and playing with onchange (NewIds). Once you save, then the values will be updated accordingly.

**Comment by pedrobaeza** [comment]
OK, but I think this can be improved adding the field to the depends list for getting automatic update even on edit mode. Is that possible or there are `inverse` methods involved?

**Comment by MiquelRForgeFlow** [comment]
@pedrobaeza your wish came true  🧞

File: account_financial_risk/models/res_partner.py [hunk 1]
```diff
@@ -108,7 +123,71 @@ class ResPartner(models.Model):
...
+        for partner in self:
+            if partner.credit_currency == "manual":
+                partner.manual_credit_currency_id = partner.risk_currency_id
+            else:
+                partner.manual_credit_currency_id = False
+
+    @api.onchange("risk_currency_id")
+    def _onchange_risk_currency_id(self):
+        for partner in self:
+            partner._compute_risk_account_amount()
```
Comment (pedrobaeza) [inline account_financial_risk/models/res_partner.py]: Can you do

```suggestion
        self._compute_risk_account_amount()
```
?

File: sale_financial_risk/models/res_partner.py [hunk 2]
```diff
@@ -37,14 +40,30 @@ def _compute_risk_sale_order(self):
...
+                company,
+                fields.Date.context_today(self),
+                round=False,
+            )
+
+    @api.onchange("risk_currency_id")
+    def _onchange_risk_currency_id(self):
+        super()._onchange_risk_currency_id()
+        for partner in self:
+            partner._compute_risk_sale_order()
```
Comment (pedrobaeza) [inline sale_financial_risk/models/res_partner.py]:
```suggestion
        self._compute_risk_sale_order()
```
?

File: account_financial_risk/models/res_partner.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply account_financial_risk/models/res_partner.py]: done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
FYI, @MiquelRForgeFlow @JordiBForgeFlow making the field `company_currency_id` stored has provoked an error very difficult to debug, but finally found and fixed removing such thing in direct commit: https://github.com/OCA/credit-control/commit/216baf808587ab08b8f02e0463a8f19e99d0b2c5

I see no real reason for such store, but if so, it should be done on other field, not reusing standard v13 one. Maybe it has been a chain of coincidences, as in v12, there wasn't such dynamic lines recomputation that needs to have immediately the field populated, and maybe is a fault of the framework not populating it on time, but the question is that it breaks standard flows. Please check why you did that way and let's discuss about it.

**Comment by pedrobaeza** [comment]
Cherry-picked for v14 in 94d9ad9a7b21c5fba63af7cbc7978e16be26cb65 and v15 in 784842163ba3db1ff68bcf166da1dd6505a59d03

**Pull request merged**